cli/src/main.rs

```diff
@@ -36,21 +44,21 @@ struct Args {
     dev: bool,
 }
 
-fn print_tree(tree: parser::Element) {
-    println!("\n{}", tree.tree_string(false));
-}
-
-fn compile_file(args: &Args) -> Result<Element, CliError> {
+fn compile_file(args: &Args) -> Result<Ast, CliError> {
     let source = fs::read_to_string(&args.input)?;
-    let document = parse(&source).unwrap();
-
     let mut ctx = Context::default();
-    let output = eval(&document, &mut ctx);
+    let output =
+        eval(&source, &mut ctx, &OutputFormat::new(&args.format)).expect("Failed to evaluate file");
 
     let mut output_file = File::create(&args.output)?;
     output_file.write_all(output.as_bytes())?;
 
-    Ok(document)
+    // Also return the Element tree for debug purposes
+    Ok(parse(&source).unwrap())
```

Comment:
Shouldn't this be `parse(&source)?` to propagate errors rather than crashing? Or is the panic caught somewhere else? Because, if I understand it correctly, the CLI will crash on parser errors

Reply:
Fixed!